.github/workflows/psalm.yml

```diff
@@ -17,81 +21,38 @@ jobs:
-            echo "COMPOSER_ROOT_VERSION=$(grep -m1 SYMFONY_VERSION .travis.yml | grep -o '[0-9.x]*').x-dev" >> $GITHUB_ENV
-
-      - name: Determine composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
-
-      - name: Cache composer dependencies
-        uses: actions/cache@v2
-        with:
-          path: ${{ steps.composer-cache.outputs.dir }}
-          key: composer-${{ github.base_ref }}
-          restore-keys: composer-
+          COMPOSER_HOME="$(composer config home)"
+          ([ -d "$COMPOSER_HOME" ] || mkdir "$COMPOSER_HOME") && cp .github/composer-config.json "$COMPOSER_HOME/config.json"
+          echo "COMPOSER_ROOT_VERSION=$(grep -m1 SYMFONY_VERSION .travis.yml | grep -o '[0-9.x]*').x-dev" >> $GITHUB_ENV
 
       - name: Install Psalm
         run: |
-          composer require psalm/phar
-          cp ./vendor/bin/psalm.phar base/psalm.phar
-          cp ./vendor/bin/psalm.phar pr/psalm.phar
-
-      - name: Install dependencies for base
-        run: |
-            cd base
-            echo "::group::modify composer.json"
-            composer remove symfony/phpunit-bridge --no-interaction --no-update
-            composer require --no-update phpunit/phpunit php-http/discovery psr/event-dispatcher
-            echo "::endgroup::"
-            echo "::group::composer update"
-            composer update --no-progress --ansi
-            echo "::endgroup::"
+          echo "::group::modify composer.json"
+          composer remove --no-update --no-interaction symfony/phpunit-bridge
+          composer require --no-update psalm/phar phpunit/phpunit php-http/discovery psr/event-dispatcher
+          echo "::endgroup::"
+          echo "::group::composer update"
+          composer update --no-progress --ansi
+          git checkout composer.json
+          echo "::endgroup::"
+          ./vendor/bin/psalm.phar --version
 
       - name: Generate Psalm baseline
         run: |
-          cd base
-          ./psalm.phar --set-baseline=.github/psalm/psalm.baseline.xml --no-progress
-
-      - name: Copy baseline
-        run: |
-          cp base/.github/psalm/psalm.baseline.xml pr/.github/psalm/psalm.baseline.xml
-
-      - name: Install dependencies for PR
-        run: |
-            cd pr
-            echo "::group::modify composer.json"
-            composer remove symfony/phpunit-bridge --no-interaction --no-update
-            composer require --no-update phpunit/phpunit php-http/discovery psr/event-dispatcher
-            echo "::endgroup::"
-            echo "::group::composer update"
-            composer update --no-progress --ansi
-            echo "::endgroup::"
-
-      - name: Cache Psalm
-        uses: actions/cache@v2
-        with:
-          path: pr/.github/psalm/cache/
-          key: psalm-${{ github.base_ref }}
-          restore-keys: psalm-
+          git checkout -m ${{ github.base_ref }}
```

Comment:
a simpler solution might be to move the `checkout PR` step after the generation of the baseline based on base

Reply:
not really: that would miss installing the deps added to the PR